vowpalwabbit/v_array.h

```diff
@@ -168,6 +168,13 @@ template<class T> void copy_array(v_array<T>& dst, const v_array<T>& src)
   push_many(dst, src._begin, src.size());
 }
 
+// use to copy arrays of types with non-trivial copy constructors, such as shared_ptr
+template<class T> void copy_array_no_memcpy(v_array<T>& dst, const v_array<T>& src)
+{ dst.erase();
```

Comment:
if you have non-trivial copy constructor, I guess you also don't have non-trivial destructors.
Is dist.erase() calling them?